ir/global.go

```diff
@@ -6,15 +6,68 @@
 package ir
 
 import (
-	"bytes"
 	"fmt"
+	"strings"
 
 	"github.com/llir/llvm/internal/enc"
 	"github.com/llir/llvm/ir/constant"
 	"github.com/llir/llvm/ir/metadata"
 	"github.com/llir/llvm/ir/types"
 )
 
+type LinkageType int
```

Comment:
In other parts of the `ir` package, types of a certain kind have been given prefixes to make them easier to find in the documentation. For example, instructions are prefixed `Inst`; e.g. [ir.InstAdd](https://godoc.org/github.com/llir/llvm/ir#InstAdd).

Thinking about it, this may not be needed for enumerates as godoc already puts enums of the same type together. Otherwise, we might have wanted to prefix these with `Linkage` or something along those lines.

Reply:
One other benefit of having a prefix is avoiding name conflicts. I did consider using a prefix for the linkage types, but couldn't find a fitting one. `Link` seemed like an obvious choice, but `LinkAvailableExternally` is nonsensical.

Reply:
Also good to see you gave the prefixing some thought. My main concern was that the documentation should be useable, as the `ir` package is rather huge. Since enums of the same type are grouped, I'm fine with keeping it as is. Name conflicts is a fair concern. However, as we are still in the `0.x` stage of development, lets solve the issue of name conflicts when and if it arise.

Reply:
At a second thought, the table of contents does benefit from added prefixes. How about `LinkageAvailableExternally`?

Reply:
At closer investigation, it turns out that enums are not part of the table of contents, so lets skip the prefix. Sorry for going back and forth :)